onboarding/src/main/kotlin/io/homeassistant/companion/android/onboarding/OnboardingNavigation.kt

```diff
@@ -78,10 +78,34 @@ internal fun NavController.navigateToOnboarding(
 }
 
 /**
- * Adds the onboarding graph to the [NavGraphBuilder]. It is dedicated to onboarding and can be accessed by using the
- * route [OnboardingRoute], the start destination of this graph is [WelcomeRoute].
+ * Defines the complete onboarding navigation graph.
  *
- * TODO update docs when the rest of the graph is ready.
+ * This graph manages the user flow from initial welcome through server connection, device
+ * registration, and optional location/security configuration. The flow adapts based on:
+ * - Server accessibility (local vs public)
+ * - App flavor (full with Google Play Services vs minimal FOSS)
+ * - Connection security (HTTP vs HTTPS)
+ * - Permission state (location access)
+ *
+ * ## Flow Overview
+ * 1. Welcome screen (only shown if [skipWelcome] is false)
+ * 2. Server discovery (only shown if [urlToOnboard] is empty)
+ * 3. Connection
+ * 3. Device naming and registration
+ * 4. Location/security configuration (conditional):
+ *    - Public servers + full flavor: Location sharing
+ *    - Local servers: Local-first explanation then location sharing (full) or security config (minimal)
+ *    - Minimal flavor + HTTP: Location for secure connection detection
+ *    - Minimal flavor + HTTPS: Complete
```

Comment:
```suggestion
 * 4. Location/security configuration (conditional)
```

This repeats the more detailed description on the functions below and risks diverging from that, so I wouldn't add a lot of detail here.